src/why3prop.ml

```diff
@@ -0,0 +1,61 @@
+(** Translation of lambdapi goals to why3 goals in propositional case *)
+
+open Terms
+open Why3
+
+(* a type to present the list of why3 constants and lambdapi terms *)
+type cnst_table = (term * Term.lsymbol) list
+
+(** [t_goal g] output the translation of a goal [g] into a why3.*)
+let rec t_goal : string -> term -> bool = fun sp g ->
+    match Basics.get_args g with
+    | (Symb({sym_name="P"; _}, _), [t])     ->
+        let (l_prop, formula) = t_prop [] [] t in
+        let symbols = List.map (fun (_, x) -> x) l_prop in
+        let tsk = Why3task.declare_symbols symbols in
+        let tsk = Why3task.add_goal tsk formula in
+        let result = Why3prover.rst (Why3prover.prover sp) tsk in
+        Why3prover.answer result.pr_answer
+    | _                                     ->
+        failwith "Goal can't be translated"
```

Comment:
There should be no failwith. Output a warning (see console.ml) and raise an error instead to stop the handling of the tactic, and catch this error in handling.ml. See tactic.ml and rewrite.ml to see how it is done in other tactics.

Reply:
Yes, use `fatal_no_pos` instead. Position will be added at the failing tactic will be added at the tactic level.

Reply:
if the tactic didn't succeed, we should return the same proof state and there is no handling for exceptions for tactics in `handle.ml`